lib/ansible/plugins/connection/docker.py

```diff
@@ -100,13 +109,23 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
             # This saves overhead from calling into docker when we don't need to
             self.actual_user = self._get_docker_remote_user()
 
+    def _split_cmd(self, argstring):
+        """Split a command string using shell semantics.
+
+        Shamelessly stolen from the SSH connection plugin.
+        """
+        return [
+            to_unicode(x.strip()) for x in shlex.split(to_bytes(argstring))
+            if x.strip()]
+
     @staticmethod
     def _sanitize_version(version):
         return re.sub('[^0-9a-zA-Z\.]', '', version)
 
     def _get_docker_version(self):
 
-        cmd = [self.docker_cmd]
+        # Force a quick copy, don't want to mutate base command.
+        cmd = [] + self.docker_cmd
```

Comment:
You should either use
```suggestion
        cmd = self.docker_cmd[:]
```
or
```suggestion
        cmd = list(self.docker_cmd)
```